### PR TITLE
Polish code

### PR DESCRIPTION
**Pull request by antoniosanct**
Hi, everyone:

Some warnings (deprecated, unused) messages cleaned up in master branch.

Regards,
Antonio.

### CODE REVIEW

File: rome/src/main/java/com/rometools/rome/io/impl/NumberParser.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public static Long parseLong(final String str) {
     public static Integer parseInt(final String str) {
         if (null != str) {
             try {
-                return new Integer(Integer.parseInt(str.trim()));
+                return Integer.valueOf(Integer.parseInt(str.trim()));
```
Comment (PatrickGotthard) [inline rome/src/main/java/com/rometools/rome/io/impl/NumberParser.java:63]: Integer.parseInt(str) is unnecessary

File: rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-
-import java.io.IOException;
-import java.net.URL;
-import java.util.HashMap;
-import java.util.Map;
-
-public class XmLReaderHttpHeaderTest {
```
Comment (PatrickGotthard) [inline rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java:17]: Wiremock is not used anymore and can be removed from pom.xml in rome-core and rome-parent

**Review by PatrickGotthard — CHANGES_REQUESTED** [review]

**Comment by PatrickGotthard** [comment]
Hi @antoniosanct,

looks good overall, but why did you replace the Alternatives-implementation?

Regards
Patrick

File: rome/src/main/java/com/rometools/rome/feed/atom/Category.java [hunk 3]
```diff
@@ -139,7 +140,7 @@ public void setSchemeResolved(final String schemeResolved) {
     }
 
     public String getSchemeResolved() {
-        return Alternatives.firstNotNull(schemeResolved, scheme);
+        return Stream.of(new String[] {schemeResolved, scheme}).filter(Objects::nonNull).findFirst().orElse(null);
```
[the remark this answers is not in the document]
Comment (antoniosanct) [reply rome/src/main/java/com/rometools/rome/feed/atom/Category.java:143]: Hi, @PatrickGotthard 
I had two main reasons: SonarQube and use of varargs in Object[]. Maybe is not so important to fix and is preferable to main Alternatives class. How do you prefer?
Regards!

File: rome/src/main/java/com/rometools/rome/io/impl/NumberParser.java (hunk 1 above)
Comment (antoniosanct) [reply rome/src/main/java/com/rometools/rome/io/impl/NumberParser.java:63]: Sure. Little mistake. I'll fix it.

File: rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java (hunk 2 above)
Comment (antoniosanct) [reply rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java:17]: OK, nice!

**Comment by antoniosanct** [comment]
Hi, @PatrickGotthard
I had two main reasons: SonarQube and use of varargs in Object[]. Maybe is not so important to fix and is preferable to main Alternatives class. How do you prefer?
Regards!

File: rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java (hunk 2 above)
Comment (PatrickGotthard) [reply rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java:17]: I just thought about it once again. We should leave the test as-is until we finally remove the URL support in XmlReader

**Review by PatrickGotthard — CHANGES_REQUESTED** [review]

**Comment by PatrickGotthard** [comment]
@antoniosanct I found the Alternatives implementation better 🙃
Even the Stream.of method you used is annotated with @SafeVarargs.

Please add @SafeVarargs to the Alternatives.firstNotNull method and replace the existing method implementation with your Stream-based approach..

File: rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java (hunk 2 above)
Comment (antoniosanct) [reply rome/src/test/java/com/rometools/rome/io/XmLReaderHttpHeaderTest.java]: Finally I restored the file.

**Review by PatrickGotthard — APPROVED** [review]

**Comment by PatrickGotthard** [comment]
Hi @antoniosanct,

looks good now, should I merge it?

Regards,
Patrick

**Comment by antoniosanct** [comment]
Hi, @PatrickGotthard 
It's a good chance in a future 2.x version. In fact, this branch has full java8-11 compatibility. In other case, I've a master_java8 branch to master, and the rest of changes (in java11 compatibility case) will be included in a future 2.x or 3.x version.
Let's wait to another opinions about it and if you want, include it when you get the best chance.
Regards,
Antonio.

**Comment by PatrickGotthard** [comment]
Am I missing something? 🤔 

All changes should be backwards-compatible so I would merge them and release version 2.1.0 soon.

**Comment by antoniosanct** [comment]
Hi, @PatrickGotthard 

It doesn't have changes on tests, then I think that this branch has backwards-compatible. In other case, how do I verify easily? With another artifact that have dependencies with ROME?

Regards,
Antonio.

**Comment by PatrickGotthard** [comment]
I have now merged the changes. There was nothing backwards-incompatible.

In summary you have:
- removed redundant interface declarations,
- removed unused imports,
- replaced deprecated code (reflection),
- and optimized the code (e.g. ```Integer.valueOf(str)``` instead of ```new Integer(str)```)

Thanks for your contribution 👍

**Comment by antoniosanct** [comment]
Hi, @PatrickGotthard 
Thanks to you for all your efforts.
Regards,
Antonio.

**Pull request merged**